internal/impl/devbox.go

```diff
@@ -746,6 +754,9 @@ func (d *Devbox) computeNixEnv(ctx context.Context, usePrintDevEnvCache bool) (m
 	}
 
 	currentEnvPath := env["PATH"]
+	if d.pure { // make nix available inside pure shell - necessary for devbox add to work
+		currentEnvPath = "/nix/var/nix/profiles/default/bin"
```

Comment:
It is! you can test it. In a pure shell, `devbox add` throws this error:
```bash
Error: We found a /nix directory but nix binary is not in your PATH and we were not 
able to find it in the usual locations. Your nix installation might be broken. 
If restarting your terminal or reinstalling nix doesn't work, please create an issue at 
https://github.com/jetpack-io/devbox/issues
```

Reply:
hm, OK. I'm not 100% sure this works in all cases (e.g. single user installs).

I just tested and single user on ubuntu installed:

```bash
$ which nix
/home/mike/.nix-profile/bin/nix
```

See https://github.com/jetpack-io/devbox/blob/main/internal/nix/source.go#L32-L32

For some reason, we are not sourcing nix correctly. In the pure shell, is `__ETC_PROFILE_NIX_SOURCED=1` set?

If that is set, then we may need a different approach.

One possible approach is to look in `$PATH` and try to find:

`/nix/var/nix/profiles/default/bin`
OR
`$HOME/.nix-profile/bin/nix` 

and keep them around. 

There's also :

```bash
xdg.StateSubpath("nix/profile/etc/profile.d/nix.sh"),
xdg.StateSubpath("nix/profiles/profile/etc/profile.d/nix.sh"),
```

I'm not sure if nix also stores bins there. 

------------

So in short, I think you need to look in path for any of these:

```bash
/nix/var/nix/profiles/default/bin
$HOME/.nix-profile/bin
xdg.StateSubpath(/nix/profile/bin)
xdg.StateSubpath(nix/profiles/profile/bin)
```

If any of those are in PATH, then they need to be retained. In practice it will only be one of them.

I'm not 100% sure there are no other nix paths, but this should cover all the cases we currently consider.